client/state/data-layer/wpcom/read/streams/index.js

```diff
@@ -228,6 +229,8 @@ export function handlePage( action, data ) {
 		...keyForPost( post ),
 		date: post[ dateProperty ],
 		...( post.comments && { comments: map( post.comments, 'ID' ).reverse() } ), // include comments for conversations
+		url: post.URL,
+		xPostMetadata: isXPost( post ) ? XPostHelper.getXPostMetadata( post ) : null,
```

Comment:
might just push the `isXPost` check down into getXPostMetadata (assuming it returns null if there's no metadata)

Reply:
Thanks for the suggestion! Moved in f18fc12.